File: src/api/models/services.py
from pydantic import BaseModel, Field


class ServiceUrls(BaseModel):
    lan: str | None = None
    tailscale: str | None = None
    public: str | None = None


class ServiceNetworkStatus(BaseModel):
    status: str | None = None


class ServiceItem(BaseModel):
    name: str
    display_name: str
    category: str
    url: str
    icon: str | None = None
    urls: ServiceUrls | None = None
    status: ServiceNetworkStatus | None = None
    enabled: bool = True


class ServiceCategory(BaseModel):
    display_name: str
    order: int = Field(default=999, ge=0)


class ServicesResponse(BaseModel):
    version: int
    categories: dict[str, ServiceCategory]
    services: list[ServiceItem]

File: frontend/src/types/services.ts
import type { NetworkMode } from './network'

export type ServiceAvailability = 'online' | 'offline' | 'unknown'

export type ServiceItem = {
  name: string
  display_name: string
  category: string
  url: string
  icon?: string | null
  urls?: {
    lan?: string | null
    tailscale?: string | null
    public?: string | null
  } | null
  status?: {
    status?: ServiceAvailability | null
  } | null
  enabled: boolean
}

export type ServiceCategory = {
  display_name: string
  order: number
}

export type ServicesResponse = {
  version: number
  categories: Record<string, ServiceCategory>
  services: ServiceItem[]
}

export type ServicesState = {
  services: ServiceItem[]
  categories: Record<string, ServiceCategory>
  loading: boolean
  error: string | null
  networkMode: NetworkMode
}
